### PR TITLE
Extract solana-account crate

### PR DESCRIPTION
**Pull request by kevinheavey**
#### Problem
Having the contents of `solana_sdk::account` available outside the SDK would help compile times and dependency hell. The most immediate unlock would be removing the solana-sdk dep from solana-account-decoder-client-types (created in #2872)

#### Summary of Changes
- move `account.rs` to its own crate
- use the new crate in solana-sdk where applicable
- re-export with a deprecation notice

### CODE REVIEW

File: sdk/account/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+documentation = "https://docs.rs/solana-account"
+version = { workspace = true }
+authors = { workspace = true }
+repository = { workspace = true }
+homepage = { workspace = true }
+license = { workspace = true }
+edition = { workspace = true }
+
+[dependencies]
+bincode = { workspace = true, optional = true }
```
Comment (kevinheavey) [inline sdk/account/Cargo.toml:13]: Making serde and bincode optional won't remove those dependencies for now as solana-program still pulls them in, but by feature gating them from the start we avoid any breaking changes. In a later PR we'll remove the solana-program dep from this new crate

**Review by febo — APPROVED** [review]
Looks good!

**Comment by kevinheavey** [comment]
Great! @yihau I've sent a crate ownership invite to anza-team

**Comment by yihau** [comment]
✅

**Pull request merged**